lib/lrama/state.rb

```diff
@@ -142,5 +160,274 @@ def rr_conflicts
         conflict.type == :reduce_reduce
       end
     end
+
+    def propagate_lookaheads(next_state)
+      next_state.kernels.map {|item|
+        lookahead_sets =
+          if item.position == 1
+            goto_follow_set(item.lhs)
+          else
+            kernel = kernels.find {|k| k.predecessor_item_of?(item) }
+            item_lookahead_set[kernel]
+          end
+
+        [item, lookahead_sets & next_state.lookahead_set_filters[item]]
```

Comment:
[note]: This is same with the definition case 1 because case 1 requires tokens which are in both lookahead_set_filters of `s'` kernel and item_lookahead_sets of `s` kernel.